game/src/core/utils/Point.java

```diff
@@ -3,38 +3,24 @@
 import core.level.utils.Coordinate;
 
 /**
- * For easy handling of positions in the dungeon. <br>
+ * A record representing a 2D point with x and y Float coordinates.
  *
- * <p>No getter needed. All attributes are public.
+ * <p>Provides utility methods for point operations such as checking distance, calculating distance,
+ * converting to a coordinate, and adding a vector.
+ *
+ * @param x The x coordinate of the point.
+ * @param y The y coordinate of the point.
+ * @see Vector2
+ * @see Coordinate
  */
-public final class Point {
-  private static final float EPSILON = 0.000001f;
-
-  /** WTF? . */
-  public float x;
-
-  /** WTF? . */
-  public float y;
-
-  /**
-   * A simple {@code float} point class.
-   *
-   * @param x the x value
-   * @param y the y value
-   */
-  public Point(float x, float y) {
-    this.x = x;
-    this.y = y;
-  }
-
+public record Point(float x, float y) {
   /**
-   * Copies the point.
+   * Create a new point from another point.
    *
-   * @param p foo
+   * @param other The other point.
    */
-  public Point(final Point p) {
-    x = p.x;
-    y = p.y;
+  public Point(Point other) {
```

Comment:
Wäre evtl. ein `static public of(Point other)` interessanter? Dann brauche ich als Kunde nur noch `Point(old)` statt `new Point(old)`? Ist aber vermutlich Geschmackssache :)